### PR TITLE
Add workflows.run support to the SDK

### PR DESCRIPTION
**Pull request by timidri**
This PR adds the `workflows.run` metadata API method which starts a workflow run.

### CODE REVIEW

**Review by hunner — APPROVED** [review]
This looks fine to me.

I couldn't figure out what's up with `tests/test_webhook.py::TestWebhookServer::test_serve_forever`. It must be something in the upstream changes since this isn't cause by in-repo changes.

File: src/relay_sdk/workflows.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+logger = logging.getLogger(__name__)
+
+
+class Workflows:
+    """Use Workflows to run a workflow with parameters"""
+
+    def __init__(self, client: Session) -> None:
+        self._client = client
+
+    def run(self, name: str, parameters: Any = {}) -> None:
```
Comment (impl) [inline src/relay_sdk/workflows.py]: The type of `parameters` should be `Mapping[str, Any]` probably.

File: README.rst [hunk 2]
```diff
@@ -52,6 +52,21 @@ method accessors.
...
+to run a workflow.
+
+Use the following to run the workflow ``my_workflow`` with parameter ``param_name`` and value ``param_value``:
+
+.. code-block:: python
+
+  from relay_sdk import Interface
+
+  relay = Interface()
+  relay.workflows.run('my_workflow', {'param_name':'param_value'})
```
Comment (impl) [inline README.rst]: You will usually want to include the name of a keyword arg when invoking such a method, e.g. `run('my_workflow', parameters={...})`.

File: src/relay_sdk/workflows.py (hunk 1 above)
Comment (timidri) [reply src/relay_sdk/workflows.py]: Done!

File: README.rst (hunk 2 above)
Comment (timidri) [reply README.rst]: Done!

File: .gitignore [hunk 3]
```diff
@@ -8,3 +8,4 @@ MANIFEST
 /.tox/
 /build/
 /dist/
+venv
```
Comment (impl) [inline .gitignore]: I don't believe this is part of any repo-specific tooling, so if this is a workflow you use across repos (e.g. how you name all virtualenvs), I would also just recommend that you create a global `.gitignore` file using e.g. something like this in your `~/.gitconfig`:

```
[core]
	excludesfile = ~/.gitignore.global
```

Comment (timidri) [reply .gitignore]: Ah, didn't realize this was possible. Done!

File: src/relay_sdk/interface.py [hunk 4]
```diff
@@ -9,6 +9,7 @@
 from .client import new_session
 from .events import Events
 from .outputs import Outputs
+from .workflows import Workflows
```
Comment (impl) [inline src/relay_sdk/interface.py]: Per linting, this import is out of order :).

File: src/relay_sdk/workflows.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+"Allows a step to run a workflow"
+import json
+import logging
+from typing import Any
+from collections.abc import Mapping
```
Comment (impl) [inline src/relay_sdk/workflows.py]: This should be imported from `typing` (i.e., added to the line above) instead for compatibility with older Python versions.

**Review by impl — APPROVED** [review]
LGTM

**Pull request merged**